skxray/core/correlation.py

```diff
@@ -402,3 +351,366 @@ def auto_corr_scat_factor(lags, beta, relaxation_rate, baseline=1):
+
+    Returns
+    -------
+    t_index : float
+        time
+    tmp_binned : array
+        matrix of correlation function without normalizations
+    pi_binned : array
+        matrix of past intensity normalizations
+    fi_binned : array
+        matrix of future intensity normalizations
+    """
+    t_index = level*num_bufs/2 + i
+
+    delay_no = (buf_no - i) % num_bufs
+
+    past_img = buf[level, delay_no]
+    future_img = buf[level, buf_no]
+
+    #  get the matrix of correlation function without normalizations
+    tmp_binned = (np.bincount(label_array,
+                              weights=past_img*future_img)[1:])
+    # get the matrix of past intensity normalizations
+    pi_binned = (np.bincount(label_array,
+                             weights=past_img)[1:])
+
+    # get the matrix of future intensity normalizations
+    fi_binned = (np.bincount(label_array,
+                             weights=future_img)[1:])
+    return t_index, tmp_binned, pi_binned, fi_binned
+
+
+def _validate_inputs(num_bufs, labels, images):
+    """
+    This is a helper function to validate inputs for both one time and
+    two time correlation
+
+    Parameters
+    ----------
+    num_bufs : int, must be even
+        maximum lag step to compute in each generation of
+        downsampling
+      labels : array
+        labeled array of the same shape as the image stack;
+        each ROI is represented by a distinct label (i.e., integer)
+    images : iterable of 2D arrays
+        dimensions are: (rr, cc)
+
+    Returns
+    -------
+    label_array : array
+        labels of the required region of interests(ROI's)
+    indices : array
+        1D array of indices into the raveled image for all
+        foreground pixels (labeled nonzero)
+        e.g., [5, 6, 7, 8, 14, 15, 21, 22]
+    num_rois : array
+        number of ROI's
+    num_pixels : array
+        number of pixels in each ROI's
+    """
+    if num_bufs % 2 != 0:
+        raise ValueError("number of channels(number of buffers) in "
+                         "multiple-taus (must be even)")
+
+    if hasattr(images, 'frame_shape'):
+        # Give a user-friendly error if we can detect the shape from pims.
+        if labels.shape != images.frame_shape:
+            raise ValueError("Shape of the images should be equal to"
+                             " shape of the labels array")
+
+    # get the pixels in each label
+    label_array, indices = roi.extract_label_indices(labels)
+
+    num_rois = np.max(label_array)
+
+    # number of pixels per ROI
+    num_pixels = np.bincount(label_array, minlength=(num_rois+1))
+    num_pixels = num_pixels[1:]
+
+    if np.any(num_pixels == 0):
+        raise ValueError("Number of pixels of the required roi's"
+                         " cannot be zero, "
+                         "num_pixels = {0}".format(num_pixels))
+
+    return label_array, indices, num_rois, num_pixels
+
+
+def one_time_from_two_time(two_time_corr):
```

Comment:
This doesn't works for multi tau, but helpful when people are doing two time correlation for all the frames. They can get the one time correlation data using this method.